server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -1351,69 +1352,75 @@ public void onFailure(Exception e) {
             @Override
             protected void doRun() {
                 final String localNodeId = clusterService.localNode().getId();
-                repository.getRepositoryData(ActionListener.wrap(repositoryData -> {
-                    try {
-                        final IndexMetadata indexMetadata = repository.getSnapshotIndexMetaData(
-                            repositoryData,
+                if (remoteStoreIndexShallowCopy == false) {
+                    executeClone(localNodeId, false);
```

Comment:
As I undersstand, except this if block, else is re-factoring. Is this correct?

Reply:
That's right. No change in the code overall. We have just moved an if condition to earlier.